easysid.cpp

```diff
@@ -26,7 +41,15 @@ EXPORT void start_capture(const char *fn)
         delete wav;
         wav = NULL;
     }
-    wav = makePcmOutputWav(fn);
+    if (pcm == NULL) {
+        if (not initAudioOutput(DEFAULT_SAMPLING_RATE)) {
```

Comment:
The sample rate is written in the WAV header. This header is written by the `PcmOutputWav` constructor. This constructor is only called from `start_capture`. So if you call `start_capture()` first, then the `PcmOutputWav` is created with the default sampling rate. Then if you change the sampling rate with `set_sampling_parameters`, then the WAV header has the wrong sample rate for the data that will later be written to it.